src/view/subgraph.rs

```diff
@@ -321,6 +310,80 @@ where
     }
 }
 
+/// An error from [Subgraph::copy_in_parent]
+#[derive(Clone, Debug, PartialEq, Eq, Error)]
+pub enum CopySubgraphError {
+    /// Tried to copy an edge crossing a subgraph boundary
+    /// when the containing graph is not a [MultiMut](crate::MultiMut)
+    #[error("Cannot copy edge between external port {external:?} and (copy of) internal port {internal:?}")]
+    CantCopyBoundary {
+        internal: PortIndex,
+        external: PortIndex,
+    },
+}
+
+impl<G: LinkMut> Subgraph<G>
+where
+    G::LinkEndpoint: Into<PortIndex>,
+{
```

Comment:
This is already a bound on `LinkEndpoint`
https://github.com/CQCL/portgraph/blob/0748032fddc7fbd057431f5de385c974af974700/src/view.rs#L283

```suggestion
impl<G: LinkMut> Subgraph<G> {
```

Reply:
And I thought I was so clever for thinking of that!! :-D